### ISSUE TITLE
[gh-aw-pilot] Nightly tt-metal L2 tests: tt-train-perf-finetuning-tests / Llama 3.2 1B on BoolQ GRPO Training [bh_p150b_civ2]: infra/flaky: jo
[Content truncated due to length]

### ISSUE THREAD

**Issue by ttmchiou** [tenstorrent/tt-auto-triage#2564]
## Summary
The scheduled "Nightly tt-metal L2 tests" run on `main` (run #9189) failed because the `tt-train-perf-finetuning-tests / Llama 3.2 1B on BoolQ GRPO Training [bh_p150b_civ2]` job hit the GitHub Actions job timeout (45 minutes) while the training/generation loop was still progressing normally — no test assertion, exception, or crash was logged. Classified as **Case 3 — likely outside tt-metal (infra/flaky/perf timeout)**.

**⚠️ AI-generated triage: may contain false positives, especially for performance regressions and timeouts.**

## Failing Test
`tt-train-perf-finetuning-tests / Llama 3.2 1B on BoolQ GRPO Training [bh_p150b_civ2]` (GRPO training/eval script under `tt-train`)

## Case 3 — Failure likely outside tt-metal
- The job log shows normal training/generation activity (reward/generation logging lines) continuing right up until the timeout; there is no assertion failure, stack trace, or crash — only the GitHub Actions annotation `##[error]The action 'Llama 3.2 1B on BoolQ GRPO Training [bh_p150b_civ2]' has timed out after 45 minutes.`
- The only commit in the diff window between the last observed run and this head commit is a one-line-per-file revert touching `tt_metal/CMakeLists.txt` and `ttnn/CMakeLists.txt` (`INSTALL_RPATH` build-dir path), unrelated to `tt-train` training/inference code paths, the SFPU/compute kernels, or the GRPO training script.
- This job is a perf/duration-sensitive long-running training benchmark (45 min ceiling); such jobs are prone to environment-dependent slowdowns (device contention, host load, network/S3 cache latency) rather than deterministic code defects.
- No other jobs in this run failed, and all core build/compile/test jobs upstream of this one passed successfully, further pointing away from a code regression.

## Failure details
- **Run**: https://github.com/tenstorrent/tt-metal/actions/runs/32338190649
- **Head commit**: [`d5f9bc4a`](https://github.com/tenstorrent/tt-metal/commit/d5f9bc4a43b1287523458f1af6a470c6b2adc180) — "Revert \"ttnn: remove non-relocatable build-dir path from INSTALL_RPATH\" (#53746)"
- **Failed jobs**: [`tt-train-perf-finetuning-tests / Llama 3.2 1B on BoolQ GRPO Training [bh_p150b_civ2]`](https://github.com/tenstorrent/tt-metal/actions/runs/32338190649/job/96333924042)
- **Error signature**:
```text
##[error]The action 'Llama 3.2 1B on BoolQ GRPO Training [bh_p150b_civ2]' has timed out after 45 minutes.
```
(preceding log lines show ongoing `[reward] gen[...]` generation output — the job was actively producing output, not hung/deadlocked, when the timeout fired.)

## Commit window analysis
Commit window is a single commit (`d5f9bc4a`), a 3-line revert limited to `tt_metal/CMakeLists.txt` and `ttnn/CMakeLists.txt` build/link path configuration.

| Commit | Description | Confidence |
|---|---|---|
| `d5f9bc4a` | Revert of INSTALL_RPATH build-dir path change (build/link config only, no runtime/training code touched) | 5 |

n/a for further disambiguation — Case 3 does not require ranking multiple suspects; this single commit is very unlikely to affect training duration/timeout behavior, supporting the infra/flaky classification.

## Who to contact
- Code owners for `tt-train/`: `ayerofieiev-tt`, `philei-tt`, `jmalone-tt` (see `.github/CODEOWNERS`)
- Author of the head commit (for context only, unlikely culprit): `blozano-tt` (https://github.com/blozano-tt)

## Recommended actions
- [ ] Re-run this job to check whether it is transient/flaky (device contention, host load, or environment latency)
- [ ] If it repeats, capture device/host utilization metrics during the run and check the S3/ccache and NOC/device queue latency around the timeout window
- [ ] Consider whether the 45-minute timeout is appropriately sized for this training benchmark relative to its typical runtime
- [ ] Escalate to CI infra owners if recurrence continues across unrelated commits

## Recurrence
First seen this run (`gh-aw-run-32338190649`).

> Generated by [CI Failure Triage (pilot)](https://github.com/tenstorrent/tt-metal/actions/runs/32349210675) · auto · 129.7 AIC · ⌖ 1.94 AIC · ⊞ 10.9K · [◷](https://github.com/search?q=repo%3Atenstorrent%2Ftt-auto-triage+is%3Aissue+%22gh-aw-workflow-call-id%3A+tenstorrent%2Ftt-metal%2Fci-failure-triage%22&type=issues)

**Comment by ttmchiou** [comment]
## New occurrence

- **Run**: https://github.com/tenstorrent/tt-metal/actions/runs/33056546111 (run #9409)
- **Date**: 2026-08-27 (workflow run for head commit `1ec5b72398130fab0331d47862cd943131570907`)
- **Failed job**: `tt-train-perf-finetuning-tests / Llama 3.2 1B on BoolQ GRPO Training [bh_p150b_civ2]`
- **Signature**: identical — `##[error]The action 'Llama 3.2 1B on BoolQ GRPO Training [bh_p150b_civ2]' has timed out after 45 minutes.` (step ran ~09:16:57–10:01:57 in this run; no assertion/exception, consistent with prior occurrences).

### Commit-window summary (this occurrence) — possible refinement to classification
- Last known-good "Nightly tt-metal L2 tests" run before this one: run #9397 (head `7632d6f67ceb2826d5caf5e33881aa07da3e3803`).
- Window: 17 commits between `7632d6f6` (exclusive) and `1ec5b723` (inclusive). Compare: https://github.com/tenstorrent/tt-metal/compare/7632d6f67ceb2826d5caf5e33881aa07da3e3803...1ec5b72398130fab0331d47862cd943131570907
- Unlike the prior occurrences on this issue (where the commit window contained no tt-train changes, supporting Case 3/infra), **this window contains exactly one tt-train/GRPO-touching commit**: `bf1375c4c23c93ef978b6c89daee25a9a53df971` — "tt-train/grpo: example cleanup, move logging & wandb init into ttml (#53433)" (author `ichovpanTT`). It directly modifies `tt-train/configs/training_configs/grpo_boolq_llama_1b_1dev.yaml` (the exact config used by this job), `tt-train/sources/ttml/trainers/grpo_trainer.py` (adds wandb init, restructures per-step timing/logging), and the boolq GRPO example scripts. None of the other 16 commits in the window touch `tt-train/**`.
- This is a plausible candidate for a genuine perf regression (not just infra flakiness) in *this* occurrence — logging/wandb-init overhead added inside or near the timed training loop could plausibly push total runtime past the 45-minute ceiling. However, I cannot run the training job in this environment (no device access) to confirm a measured before/after step-time delta, so I am not elevating this to a confident Case 1 override of the existing Case 3 classification — flagging it here as a specific, testable hypothesis for whoever investigates next.
- Confidence: `bf1375c4` ~55 (plausible, unconfirmed without timing data) — all other 16 commits in window ruled out as unrelated subsystems (Qwen/Llama sampling, repeat_interleave codegen, UMD bump, fused-test runtime cuts, zeroing API, workflow-only, expand op, PCH coverage, dispatch/MetalContext threading, topology mapper, argmax/pad metal2.0 ports, ttsim bump, GroupNorm, sweep_rank_binding fix, fabric routing golden).

### Who to contact (for this hypothesis)
- `ichovpanTT` (author of `bf1375c4`) — https://github.com/ichovpanTT
- tt-train code owners: `ayerofieiev-tt`, `philei-tt`, `jmalone-tt`

**⚠️ AI-generated analysis — may contain false positives/inaccuracies, especially for performance regressions and timeouts. The `bf1375c4` hypothesis is unconfirmed and should be validated with actual timing measurements before acting.**

gh-aw-run-33056546111

> Generated by [CI Failure Triage (pilot)](https://github.com/tenstorrent/tt-metal/actions/runs/33072033733) · auto · 426.2 AIC · ⌖ 2.26 AIC · ⊞ 11.1K · [◷](https://github.com/search?q=repo%3Atenstorrent%2Ftt-auto-triage+%22gh-aw-workflow-call-id%3A+tenstorrent%2Ftt-metal%2Fci-failure-triage%22&type=issues)

**Comment by ttmchiou** [comment]
### Recurrence — run 33157823347

- **Date**: this run (head `26f496cfbbd3b115ce961ed1decf2df0146c9f96`)
- **Run**: https://github.com/tenstorrent/tt-metal/actions/runs/33157823347
- **Failed job**: `tt-train-perf-finetuning-tests / Llama 3.2 1B on BoolQ GRPO Training [bh_p150b_civ2]`
- **Signature**: identical — GitHub Actions step timeout after 45 minutes, no assertion/exception in the log.
- **Commit window**: not independently recomputed for this occurrence; consistent with the original Case 3 (infra/flaky) classification — no evidence of a code-level regression found in the training/inference paths.

> ⚠️ AI-generated triage; may contain false positives, especially for performance regressions and timeouts.

gh-aw-run-33157823347

> Generated by [CI Failure Triage (pilot)](https://github.com/tenstorrent/tt-metal/actions/runs/33168488822) · auto · 271.2 AIC · ⌖ 2.3 AIC · ⊞ 11.1K · [◷](https://github.com/search?q=repo%3Atenstorrent%2Ftt-auto-triage+%22gh-aw-workflow-call-id%3A+tenstorrent%2Ftt-metal%2Fci-failure-triage%22&type=issues)

**Comment by ttmchiou** [comment]
## Recurrence — run 33363013366

- **Run**: https://github.com/tenstorrent/tt-metal/actions/runs/33363013366
- **Date**: 2026-08-31
- **Head commit**: `3a33073484f7fd2df67f4731d3c4d90907cf7b00` (#51217: Fix shared L1 scratch race in unaligned reshard same-width (#53028))
- **Commit window**: since last successful run (#9438, head `9dd07775dc646ec767754e2823bfcd60a1f0a195`), ~46 commits, none touching `tt-train` GRPO/perf-finetuning training code or the timeout ceiling itself.
- **Signature match**: identical — job `tt-train-perf-finetuning-tests / Llama 3.2 1B on BoolQ GRPO Training [bh_p150b_civ2]` hit the 45-minute GitHub Actions timeout again (`##[error]The action ... has timed out after 45 minutes.`), no assertion/crash.
- This continues to look like Case 3 (infra/flaky perf-duration timeout), consistent with the existing classification.

gh-aw-run-33363013366

⚠️ AI-generated triage — may contain false positives, especially for performance regressions and timeouts.

> Generated by [CI Failure Triage (pilot)](https://github.com/tenstorrent/tt-metal/actions/runs/33383630733) · auto · 289.6 AIC · ⌖ 2.18 AIC · ⊞ 11.1K · [◷](https://github.com/search?q=repo%3Atenstorrent%2Ftt-auto-triage+%22gh-aw-workflow-call-id%3A+tenstorrent%2Ftt-metal%2Fci-failure-triage%22&type=issues)

**Comment by ttmchiou** [comment]
### Recurrence — run 33597464570

**Run**: https://github.com/tenstorrent/tt-metal/actions/runs/33597464570 (run #9519)
**Date**: 2026-09-02 (head commit `91254f3ba2c155db827515b0369c014c2134cb90`)
**Job**: `tt-train-perf-finetuning-tests / Llama 3.2 1B on BoolQ GRPO Training [bh_p150b_civ2]`
**Signature**: identical — `##[error]The action has timed out.` after 45 minutes; no assertion or exception recorded, training/generation logging still active at time of timeout.

**Commit window**: last successful run of this workflow on main is run #9438 (id 33172891508, head `9dd07775dc646ec767754e2823bfcd60a1f0a195`). The window to this run's head spans ~100+ commits over roughly a week; no single commit stands out as plausibly affecting this training job's runtime — consistent with Case 3, which does not require commit attribution.

This is at least the 3rd recorded occurrence of this same infra/timeout signature on this job — reinforces the infra/flaky classification rather than a code regression.

⚠️ AI-generated analysis; may contain false positives, especially for timeouts.

gh-aw-run-33597464570

> Generated by [CI Failure Triage (pilot)](https://github.com/tenstorrent/tt-metal/actions/runs/33609798968) · auto · 351.2 AIC · ⌖ 2.39 AIC · ⊞ 11.1K · [◷](https://github.com/search?q=repo%3Atenstorrent%2Ftt-auto-triage+%22gh-aw-workflow-call-id%3A+tenstorrent%2Ftt-metal%2Fci-failure-triage%22&type=issues)

**Comment by ttmchiou** [comment]
## Recurrence — run 33721785533 (run #9555, 2026-09-03)

Same failure signature reproduced again: `tt-train-perf-finetuning-tests / Llama 3.2 1B on BoolQ GRPO Training [bh_p150b_civ2]` hit the 45-minute GitHub Actions job timeout mid-training (step 51 of the training loop was in progress when the timeout fired — no assertion, exception, or crash logged). Identical to the signature already tracked here and in #2567.

**Commit window**: last known-successful run of this workflow before this one was run #9438 (head `9dd07775dc646ec767754e2823bfcd60a1f0a195`, 2026-08-28). Compare: https://github.com/tenstorrent/tt-metal/compare/9dd07775dc646ec767754e2823bfcd60a1f0a195...b36d9d43206c92e742984e92f2ef5d2c06df0747 — this window is large (~9 nightly runs / several days), and none of it changed the underlying classification: this is still a duration/perf-sensitive training benchmark timing out, not a code assertion. No single commit stands out as a plausible culprit for a timing regression in this window based on file-path review; treating this recurrence as the same **Case 3 (infra/flaky, perf-timeout)** bucket as the original issue.

- **Run**: https://github.com/tenstorrent/tt-metal/actions/runs/33721785533
- **Head commit**: b36d9d43206c92e742984e92f2ef5d2c06df0747
- **Signature identical**: yes (same job, same timeout-with-no-assertion pattern)

This is now at least 3 occurrences (run #9189/32338190649 in #2564 originally, run #9189 also in #2567, and this run #9555). Recommend escalating to check whether this benchmark's 45-minute ceiling is simply too tight for current hardware/queue conditions, rather than continuing to re-file as a one-off timeout each time.

⚠️ AI-generated triage — may contain false positives, especially for performance regressions and timeouts.

gh-aw-run-33721785533

> Generated by [CI Failure Triage (pilot)](https://github.com/tenstorrent/tt-metal/actions/runs/33739726680) · auto · 361.3 AIC · ⌖ 3.26 AIC · ⊞ 11.1K · [◷](https://github.com/search?q=repo%3Atenstorrent%2Ftt-auto-triage+%22gh-aw-workflow-call-id%3A+tenstorrent%2Ftt-metal%2Fci-failure-triage%22&type=issues)
